### PR TITLE
Shared config

### PR DESCRIPTION
**Pull request by jmeekhof**
Pull request for issue #685

### CODE REVIEW

File: deploy/lib/server_config.rb [hunk 1]
```diff
@@ -2685,8 +2685,9 @@ def ServerConfig.properties(prop_file_location = @@path)
     properties = ServerConfig.load_properties(default_properties_file, "ml.")
     properties.merge!(ServerConfig.load_properties(properties_file, "ml."))
 
+    #Look for optional shared_config, if it is set grab the properties from path relative to the root of the roxy project
     if properties['ml.shared_config'] then
```
Comment (RobertSzkutak) [inline deploy/lib/server_config.rb]: I'm thinking we're also going to need to do these tests :

1) Does a shared properties file exist? (Could aid debugging)
2) Should a shared properties file be used for this environment? My understanding of your use case is that there's inevitably going to end up being two shared properties files applied to different environments respectively. So we'll need a mechanism to map environment properties to shared properties appropriately.

File: deploy/lib/server_config.rb [hunk 2]
```diff
@@ -2685,8 +2685,9 @@ def ServerConfig.properties(prop_file_location = @@path)
...
-      shared_properties_file = ServerConfig.expand_path(properties['ml.shared_config'])
+      shared_properties_file = ServerConfig.expand_path("#{@@path}/../#{properties['ml.shared_config']}")
```
Comment (RobertSzkutak) [inline deploy/lib/server_config.rb]: I'm assuming we need to make the path configurable on this to fulfill your need to handle this via a submodule?

**Comment by RobertSzkutak** [comment]
Great start! Will probably send you a PR this evening if you don't beat me to it..

File: deploy/lib/server_config.rb (hunk 1 above)
Comment (jmeekhof) [inline deploy/lib/server_config.rb]: We could certainly add a file existence test, but that could cause issues with figuring out *if*  the shared config file is working. In other words, if the shared config file doesn't exist, and we're trying to use it, and the code silently jumps over the missing file, how do you know you've made a mistake? Sorry for the run on sentence.

Comment (RobertSzkutak) [inline deploy/lib/server_config.rb]: @jmeekhof I'm totally good with this after speaking with you about the use case.

**Comment by RobertSzkutak** [comment]
Sent you a PR @jmeekhof with trivial documentation. If that looks good to you, I'll tap someone else to review before merging.

**Comment by jmeekhof** [comment]
@RobertSzkutak I got your pull request, merged your change, and added a small comment of my own.

**Comment by RobertSzkutak** [comment]
Looks good to me! @dmcassel or @grtjn could you also give this a quick review? Thanks!

**Comment by dmcassel** [comment]
One small change and then I'm good for this to be merged:

> Config file loction is relative to the root of your roxy directory

Please change "roxy directory" to "deploy directory". For people using the MVC application structure, there is a src/roxy/ directory, so this could be confusing. (Might as well fix the "loction" typo at the same time.)

**Comment by jmeekhof** [comment]
@dmcassel The way this is currently coded, its looking in the "root" of the application, not the deploy directory. When you say "deploy" directory, I think "deploy/". 

What is the best wording to use to avoid confusion, as I think "deploy" is also a bit confusing as well.

**Comment by dmcassel** [comment]
In that case, I would refer to "the project root".

**Comment by jmeekhof** [comment]
@dmcassel, @RobertSzkutak I've updated the wording to avoid inevitable confusion. I think it's ready to be merged!

**Comment by RobertSzkutak** [comment]
Fine for me! Merging..

**Pull request merged**